http_parser.c

```diff
@@ -1007,7 +1007,7 @@ size_t http_parser_execute (http_parser *parser,
           UPDATE_STATE(s_req_spaces_before_url);
         } else if (ch == matcher[parser->index]) {
           ; /* nada */
-        } else if (IS_ALPHA(ch)) {
+        } else if ((ch >= 'A' && ch <= 'Z') || ch == '-') {
```

Comment:
It does, just for the check. It does not modify `ch` for later code.

But if the letter is lowercase instead of uppercase, it will (when mixed with the other variables for the fancy select value) never match any case statement. It will always hit the "default" case. Which is the same as not matching this if condition.